k8s/production/custom/prune-buildcache/service-accounts.yaml

```diff
@@ -0,0 +1,8 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: prune-buildcache
+  namespace: custom
+  annotations:
+    eks.amazonaws.com/role-arn: arn:aws:iam::588562868276:role/DeleteAndPutObjectsFromBucketSpackBinaries
```

Comment:
I think this service account would be better off as a terraform `kubectl_manifest`, as we could then just reference this IAM role with TF instead of duplicating the ARN here.

Reply:
Fixed in https://github.com/spack/spack-infrastructure/pull/415/commits/f5ebc95d9270d0f8e5f330654c4840c51eaa3565